tests/unit/select_connection_ioloop_tests.py

```diff
@@ -33,8 +39,10 @@ def tearDown(self):
         self.ioloop.remove_timeout(self.fail_timer)
         self.ioloop = None
 
-    def start(self):
-        self.fail_timer = self.ioloop.add_timeout(self.TIMEOUT, self.on_timeout)
+    def start(self, callback = None):
```

Comment:
Let's revert this change (callback ...) in `start`. It's not necessary, and breaks the intended behavior of `raise AssertionError('Test timed out')` when the test times out. I will explain why it's not necessary in other comments in this code.